funcframework/framework.go

```diff
@@ -96,11 +97,50 @@ func RegisterCloudEventFunctionContext(ctx context.Context, path string, fn func
 	return registerCloudEventFunction(ctx, path, fn, handler)
 }
 
+// Declaratively registers a HTTP function.
+func HTTP(name string, fn func(http.ResponseWriter, *http.Request)) {
+	defer recoverPanic("Registration panic")
+
+	// Register the function.
+	registry.RegisterHTTP(name, fn)
+}
+
+// Declaratively registers a CloudEvent function.
+func CloudEvent(name string, fn func(context.Context, cloudevents.Event) error) {
+	defer recoverPanic("Registration panic")
+
+	// Register the function.
+	registry.RegisterCloudEvent(name, fn)
+}
+
 // Start serves an HTTP server with registered function(s).
 func Start(port string) error {
+	// If FUNCTION_TARGET, try to start with that registered function
```

Comment:
It would be good to have a unit test in `framework_test.go` that tested what happens when Register* calls get mixed with declaration function registration. 

Could you try adding that? You could pull out this function-selecting logic into a `overrideHandlerWithRegisteredFunctions(h handler)` or something helper func to help with testing. 

I imagine it would be kind of similar to one of the existing tests, except using the handler

```
tests := []struct{
     registerFn func() error
     funcTarget string
} {
   {
     registerFn: func() {
         
		if err; funcframework.Register(..., func {return error} ) {
                     return err;
                 }
                //etc, etc
		funcframework.Http("actuallyCallThisOne", func {return non-error})
		funcframework.Http(..., func {return error})
    },
    funcTarget: "actuallyCallThisOne", 
    }
}
for _, tc := range tests {
                hander = http.NewServeMux()
		ctx := context.Background()

                if err := tc.registerFn(); err != nil {
                    ...
                }
		os.SetEnv("FUNCTION_TARGET", tc.funcTarget)
		overrideHandlerWithRegisteredFunctions(handler)

		srv := httptest.NewServer(handler)
		defer srv.Close()

		req, err := http.NewRequest("POST", srv.URL, bytes.NewBuffer(tc.body))
                if err != nil {
                ...
                }
}
```

Reply:
After a bit of work, I've added a unit test for DFS functions here.